src/onlyspans.polyleads.ui/src/components/login/AuthForm.tsx

```diff
@@ -1,59 +1,121 @@
+      .min(4, 'Имя должно содержать не менее 4 символов'),
+    password: z
+      .string({ required_error: 'Поле должно быть заполнено' })
+      .min(6, 'Пароль должен содержать не менее 6 символов'),
+  });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+
   return (
-    <div className={'grid gap-6'}>
-      <form onSubmit={vm.login}>
-        <div className='grid gap-2'>
-          <div className='grid gap-1'>
-            <Input
-              id='email'
-              minLength={4}
-              required={true}
-              placeholder='Имя пользователя или почта'
-              autoComplete='email'
-              autoCorrect='off'
-              disabled={vm.isLoading}
-            />
-          </div>
-          <div className='grid gap-1'>
-            <Input
-              id='password'
-              minLength={6}
-              required={true}
-              placeholder='Пароль'
-              autoComplete='off'
-              autoCorrect='off'
-              disabled={vm.isLoading}
-            />
-          </div>
-          <Button disabled={vm.isLoading}>
-            {vm.isLoading ? (
-              <>
-                <LoaderCircle className='mr-2 h-4 w-4 animate-spin' />
-                Подождите
-              </>
-            ) : (
-              'Войти'
-            )}
-          </Button>
-        </div>
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(() => vm.login)}
+        className='grid gap-2'
+      >
+        <FormField
+          control={form.control}
+          name='email'
+          render={({ field }) => (
+            <FormItem>
+              <FormControl>
+                <Input
+                  id='email'
+                  placeholder='Имя пользователя или почта'
+                  autoComplete='email'
+                  autoCorrect='off'
+                  disabled={vm.isLoading}
+                  {...field}
+                />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name='password'
+          render={({ field }) => (
+            <FormItem>
+              <FormControl>
+                <div className='relative'>
+                  <Input
+                    id="password"
+                    type={vm.isPasswordShown ? 'text' : 'password'}
+                    placeholder='Пароль'
+                    autoComplete="new-password"
+                    autoCorrect="off"
+                    disabled={vm.isLoading}
+                    {...field}
+                  />
+                  <Button
+                    type='button'
+                    variant='ghost'
+                    className='absolute top-0 right-0 px-3 py-2 hover:bg-transparent'
+                    onClick={() => vm.setPasswordShown(!vm.isPasswordShown)}
```

Comment:
```suggestion
                    onClick={vm.togglePasswordShown}
```

внутри уже используешь vm.isPasswordShown